Wikipedia/Code/WikipediaURLTranslations.swift

```diff
@@ -0,0 +1,93 @@
+
+/*
+ TODO:
+ - re-export the json with all strings lower-cased then update the "commonNamespace" func to force "namespaceString" to lower-case
+ - replace spaces in the json with "_" too - also replace them in "namespaceString"
+*/
+
+struct WikipediaURLTranslations:Codable {
+    private var languagecode:Dictionary<String, WikipediaURLLanguageCodeTranslations> = Dictionary()
+    private static let sharedLookupTable = WikipediaURLTranslations.init(fileName: "wikipedia-namespaces").languagecode
+    private init(fileName : String) {
+        guard
+            let url = Bundle.main.url(forResource: fileName, withExtension: "json"),
+            let data = try? Data(contentsOf: url)
+        else {
+            assertionFailure("Unable to open JSON")
+            return
+        }
+        do {
+            languagecode = (try JSONDecoder().decode(WikipediaURLTranslations.self, from: data)).languagecode
+        } catch {
+            assertionFailure("Unable to decode WikipediaURLTranslations from JSON data")
+        }
+    }
+}
+
+extension WikipediaURLTranslations {
+    static func commonNamespace(for namespaceString: String, in languageCode: String) -> WikipediaURLCommonNamespace? {
+        return WikipediaURLTranslations.sharedLookupTable[languageCode]?.namespace[namespaceString]
+    }
+
+    static func mainpage(in languageCode: String) -> String? {
+        return WikipediaURLTranslations.sharedLookupTable[languageCode]?.mainpage
+    }
+}
+
+struct WikipediaURLLanguageCodeTranslations:Codable {
+    let namespace:Dictionary<String, WikipediaURLCommonNamespace>
+    let mainpage:String
```

Comment:
I'm generally not a stickler for the more nuanced style questions, but all of the Swift documentation has a space between the `:` and the type